packages/shared-ux/button_toolbar/src/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { ReactElement } from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { IconButtonGroup, PrimaryButton } from '../buttons';
+import { ToolbarPopover } from '../popover';
+
+/** type for cases with both button or a popover could be used */
+export type Button = typeof PrimaryButton | typeof ToolbarPopover;
+
+/** Specific type for the toolbar children in its props */
+interface NamedSlots {
+  primaryButton: ReactElement<Button>;
+  iconButtonGroup?: ReactElement<typeof IconButtonGroup>;
+  extraButtons?: Array<ReactElement<Button>> | undefined;
+}
+
+/**
+ * Props for a generic toolbar component
+ */
+export interface Props {
+  children: NamedSlots;
+}
+
+/**
+ *
+ * @param children of the toolbar such as a popover or button
+ * @returns Toolbar component
+ */
+export const Toolbar = ({ children }: Props) => {
+  const { primaryButton, iconButtonGroup, extraButtons = [] } = children;
+
+  if (extraButtons.length > 120) {
+    throw new Error(
+      'There are over 120 extra buttons. Please consider limiting the number of buttons.'
```

Comment:
Should the error be localized?

Reply:
Probably, yes! Thank you! [3247796](https://github.com/elastic/kibana/pull/137224/commits/324779666ce1e58346ae84b3ef6ac4d08d7ecdbc) I think I did it right? 🤞🏻